nc-news_/src/Components/ArticleList/ListProvider.jsx

```diff
@@ -0,0 +1,48 @@
+import apiClient from "../../api";
+import { useEffect, useState } from "react";
+import ArticleCard from "./ArticleCard";
+
+const ListProvider = () => {
+  const [articleList, setArticleList] = useState([]);
+  const [page, setPage] = useState(1);
+  const [articleCount, setArticleCount] = useState(0);
+  const totalPages = Math.ceil(articleCount / 10);
+
+  const nextPageHandler = () => {
+    setPage(page + 1);
+  };
+  const previousPageHandler = () => {
+    setPage(page - 1);
+  };
+
+  useEffect(() => {
```

Comment:
You could add some loading state here to conditionally render something to tell a user the data is still being fetched while they are waiting ( for those on a slower internet connection ) 💡